Tasks/SonarQubePostTest/CodeAnalysisFilePathComputation.ps1

```diff
@@ -0,0 +1,170 @@
+$ProjectGuidAndFilePathMap = @{}
+$ComponentKeyAndPathMap = @{}
+$ComponentKeyAndRelativePathCache = @{}
+
+function ConstructComponentKeyAndPathMap($json)
+{
+    foreach ($component in $json.Components)
+    {
+        #Write-Host "Component Key: $($component.key), Path:$($component.path)"
+        $ComponentKeyAndPathMap.Add($component.key, $component.path)
+    }
+}
+
+#returns a path relative to the repo root for a file which has new code analysis issue(s)
+function GetRelativeFilePath($component)
+{
+    if ($ComponentKeyAndRelativePathCache.ContainsKey($component))
+    {
+        $relativeFilePath = $ComponentKeyAndRelativePathCache[$component]
+        Write-Verbose -Verbose "GetRelativeFilePath: Found cached entry, returning data from cache, relativePath:$relativeFilePath"
```

Comment:
As discussed during the review, can you ask TFS Build Dogfooding about guidance in logging?